src/auth/cephx/CephxProtocol.cc

```diff
@@ -584,7 +584,7 @@ bool CephXAuthorizer::add_challenge(CephContext *cct,
   auto p = challenge.begin();
   if (!p.end()) {
     std::string error;
-    CephXAuthorizeChallenge ch;
+    CephXAuthorizeChallenge ch = CephXAuthorizeChallenge();
```

Comment:
i don't understand this change. isn't this equivalent to calling its default constructor?

Reply:
The variable 'ch' is being used without being initialized.

Reply:
thanks. it's `ch.server_challenge` that's uninitialized by its default constructor. wouldn't `ch = CephXAuthorizeChallenge()` just assign some uninitialized value to it?

you could use braces to zero-initialize it explicitly: `CephXAuthorizeChallenge cx{0};`

Reply:
> thanks. it's `ch.server_challenge` that's uninitialized by its default constructor. wouldn't `ch = CephXAuthorizeChallenge()` just assign some uninitialized value to it?

this is not quite true. i think `ch = CephXAuthorizeChallenge()`  is value initialization. since `CephXAuthorizeChallenge` has a default construct. the object is zero-initialized in this case.

while 
```c++
CephXAuthorizeChallenge ch;
```

is default initialized. to the value of `ch.server_challenge` is indeterminate.

> 
> you could use braces to zero-initialize it explicitly: `CephXAuthorizeChallenge cx{0};`

another way is probably just put
```c++
CephXAuthorizeChallenge ch{};
```

to call zero initialization explicitly.

Reply:
thanks @tchaikov, i agree that `CephXAuthorizeChallenge ch = CephXAuthorizeChallenge();` will zero-initialize the `server_challenge` member after reading https://en.cppreference.com/w/cpp/language/value_initialization